engine/src/main/java/io/zeebe/engine/processing/bpmn/behavior/BpmnEventSubscriptionBehavior.java

```diff
@@ -137,12 +139,18 @@ private long triggerBoundaryEvent(
       deferActivatingEvent(context, boundaryElementInstanceKey, record);
 
       stateTransitionBehavior.transitionToTerminating(context);
+      return boundaryElementInstanceKey;
 
     } else {
-      publishActivatingEvent(boundaryElementInstanceKey, record);
+      // copy temporary variables here because they are read in publishActivatingEvent()
+      if (eventTrigger.getVariables() != null && eventTrigger.getVariables().capacity() > 0) {
+        variablesState.setTemporaryVariables(
+            boundaryElementInstanceKey, eventTrigger.getVariables());
+      }
+      publishActivatingEvent(context, boundaryElementInstanceKey, record);
+      // avoid creating temporary variables again - variables are already passed in
+      return -1L;
```

Comment:
The passing of the event variables and the usage of the temporary variables are not great. However, it was the easiest way without changing too much. In the end, this part will be replaced by `findEventTrigger()` + `activateTriggeredEvent()`.